src/wallet/coin_selection.rs

```diff
@@ -567,27 +571,28 @@ impl BranchAndBoundCoinSelection {
         for _ in 0..BNB_TOTAL_TRIES {
             // Conditions for starting a backtrack
             let mut backtrack = false;
-            // Cannot possibly reach target with the amount remaining in the curr_available_value,
+            // Cannot possibly reach target with the amount remaining in the available_value,
             // or the selected value is out of range.
             // Go back and try other branch
-            if curr_value + curr_available_value < target_amount
-                || curr_value > target_amount + cost_of_change as i64
+            if selected_value + available_value < target_amount
+                || selected_value > target_amount + cost_of_change as i64
             {
                 backtrack = true;
-            } else if curr_value >= target_amount {
+            } else if selected_value >= target_amount {
                 // Selected value is within range, there's no point in going forward. Start
                 // backtracking
                 backtrack = true;
 
                 // If we found a solution better than the previous one, or if there wasn't previous
                 // solution, update the best solution
-                if best_selection_value.is_none() || curr_value < best_selection_value.unwrap() {
+                if best_selection_value.is_none() || selected_value < best_selection_value.unwrap()
+                {
                     best_selection = current_selection.clone();
-                    best_selection_value = Some(curr_value);
+                    best_selection_value = Some(selected_value);
                 }
 
                 // If we found a perfect match, break here
-                if curr_value == target_amount {
+                if selected_value == target_amount {
                     break;
                 }
```

Comment:
This may not necessarily be the solution with the best waste score (but of course an optimal solution if minimizing the excess).